pkg/apis/controller/suggestions/v1beta1/suggestion_types.go

```diff
@@ -78,12 +78,15 @@ type TrialAssignment struct {
 	// Suggestion results with Trial parameters
 	ParameterAssignments []common.ParameterAssignment `json:"parameterAssignments,omitempty"`
 
-	//Name of the suggestion
+	// Name of the suggestion
 	Name string `json:"name,omitempty"`
 
 	// Rules for early stopping techniques
 	// Contains rule name, value and comparison type
 	EarlyStoppingRules []common.EarlyStoppingRule `json:"earlyStoppingRules,omitempty"`
+
+	// Suggestion metadata to attach to Trial job
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
You are correct about uid and generation annotations being used by the suggestion service to track a spawned trial to completion! I'm not sure I follow the algorithmSettings solution - isn't that field static once the Suggestion is created?

Reply:
Thanks for the notes, it's a bit clearer now! I see how we could transmit the uid/gen via AlgorithmSettings, but once the experiment is updated, that information is not stored in a [Trial](https://github.com/kubeflow/katib/blob/280eaed8cc27e25241fdcd0517354ed5deb732d9/pkg/apis/manager/v1beta1/api.proto#L209). So if we have the `trial-1-uid: <Suggestion defined uid>` map in the Experiment, how might we tie that to the backend Trial uid that gets defined on spawn so we can link the two in the Suggestion service?

Reply:
I guess, if your Suggestion can define Trial name, you can link Trials and `generation`, `parent` data from `AlgorithmSettings`, because you are using Trial annotations only in PBT service: https://github.com/kubeflow/katib/blob/ff3da32328d1c119d98236943bfb79b07ead5207/pkg/suggestion/v1beta1/pbt/service.py#L295

But it looks tricky and it ties to my original question: https://github.com/kubeflow/katib/pull/1833#discussion_r865033485.
Do we want to set Trial name from Suggestion service ?

Reply:
I agree that setting the name from the Suggestion service would be a cleaner flow to ensure we can track the Trial. Assuming we go with that, what is the preference for passing other metadata we want tied to the Trial (e.g. generation)? Is the Annotation field okay or should I change that to Label?

Reply:
I prefer `Labels`, since [Kubernetes docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects) say that Lables are used to select objects. Also, I think that Labels should go to metadata so we can use [Kubernetes Label Selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to select Trials for UI, etc.

Let's see what others say.
@gaocegege @johnugeorge @tenzen-y ?

Reply:
I have a question @andreyvelich @a9p.

Are these trial settings (`Labels` or `Annotations`) only referenced from Katib components (e.g., Katib UI, Suggestion Service)? Or are they expected end-users to use (e.g., select object)?

If used only in Katib components, I think it would be better `Annotations`.

Since the labels are propagated to the trial pod, it might reduce the visibility of `kubectl` results (e.g. `kubectl get pods -n kubeflow --show-labels`) to add many labels not to be used by end-users.

Reply:
The annotations are now only used to tie information to the Trial for the UI. I wasn't imagining users parsing through those fields for PBT at least.

Reply:
What if users want to compare all PBT Trials with `generation` 2 ? What do others prefer (`Labels` or `Annotations`) @gaocegege @johnugeorge ?
Do we have any examples from other CRDs when they store object metadata from various services ?

Reply:
As per docs, annotations contain non identifiable info which doesn't need selection or watch.  As @andreyvelich pointed out, won't they be used for selection/filter ?

Reply:
It might be better to use `Labels` in preference to Kubernetes docs rather than the visibility of kubectl results.
Sorry if I have confused you.